user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -139,12 +149,111 @@ protected void createOrUpdate(Reconciliation reconciliation, KafkaUser kafkaUser
             scramAcls = user.getSimpleAclRules();
         }
 
+        KafkaUserStatus userStatus = new KafkaUserStatus();
         CompositeFuture.join(
                 scramShaCredentialOperator.reconcile(user.getName(), password),
-                secretOperations.reconcile(namespace, user.getSecretName(), desired),
+                reconcileSecretAndSetStatus(namespace, user, desired, userStatus),
                 aclOperations.reconcile(KafkaUserModel.getTlsUserName(userName), tlsAcls),
                 aclOperations.reconcile(KafkaUserModel.getScramUserName(userName), scramAcls))
-                .map((Void) null).setHandler(handler);
+                .setHandler(reconciliationResult -> {
+
+                    if (kafkaUser.getMetadata().getGeneration() != null)    {
+                        userStatus.setObservedGeneration(kafkaUser.getMetadata().getGeneration());
+                    }
+
+                    Condition readyCondition = ConditionUtils.buildConditionFromReconciliationResult(reconciliationResult.mapEmpty());
```

Comment:
Can we call `setCondition` right after we get it? I.e. have the lines 164 and 167 next to each other? The code will be easier to read.